### PR TITLE
Fix generators with default args

### PR DESCRIPTION
**Pull request by gzzhanghao**
See https://phabricator.babeljs.io/T2776

### CODE REVIEW

**Comment by hzoo** [comment]
Awesome!

What do you think about doing an `exec.js` test like in https://github.com/frantic/babel/blob/ca1c4bba868baeaae783a4155b6c8beec8774208/packages/babel-plugin-transform-react-jsx-source/test/fixtures/react-source/basic-sample/exec.js to assert certain values in like in original issue where `test.start();` was expected to be 1.

**Comment by gzzhanghao** [comment]
@hzoo added :)

File: packages/babel-traverse/src/context.js [hunk 1]
```diff
@@ -105,6 +105,9 @@ export default class TraversalContext {
       path.resync();
       path.pushContext(this);
 
+      // this path is no longer belongs to the tree
```
Comment (fabiomcosta) [inline packages/babel-traverse/src/context.js]: nit: this path is no longer... -> this path no longer...

File: packages/babel-traverse/src/context.js [hunk 2]
```diff
@@ -105,6 +105,9 @@ export default class TraversalContext {
...
+      if (path.key === null) continue;
```
Comment (amasad) [inline packages/babel-traverse/src/context.js]: What causes this to be set to null? we might want to address that instead of doing it here

Comment (gzzhanghao) [inline packages/babel-traverse/src/context.js]: @amasad It was set to null here:
https://github.com/babel/babel/blob/master/packages/babel-traverse/src/path/context.js#L165

And we can reproduce it with two steps:

1. insert a path into a upper level queue (VariableDeclarationStatement was inserted after the FunctionDeclarationStatement)
2. replace the path's parent path (FunctionDeclarationStatement was replaced with a CallExpression)

But I can't figure out why the container of inserted path is not modified after the _containerInsert operation. Any idea?

**Pull request merged**